### PR TITLE
add support for alternate settings

### PR DESCRIPTION
**Pull request by kiffie**
Here is a PR based on the two code snippets from issue "Alternate setting support #49" from @redpfire and myself. @ianrrees, you might want to have a look at it.

 @mvirkkunen, what do you think about it? Your comments are appreciated. If you could merge it, we could get rid of all the personal forks.

### CODE REVIEW

File: src/descriptor.rs [hunk 1]
```diff
@@ -175,24 +175,64 @@ impl DescriptorWriter<'_> {
...
+    /// * `number` - Interface number previously allocated with
+    ///   [`UsbBusAllocator::interface`](crate::bus::UsbBusAllocator::interface).
+    /// * `alternate_setting` - number of the alternate setting
+    /// * `interface_class` - Class code assigned by USB.org. Use `0xff` for vendor-specific devices
+    ///   that do not conform to any class.
+    /// * `interface_sub_class` - Sub-class code. Depends on class.
+    /// * `interface_protocol` - Protocol code. Depends on class and sub-class.
+    /// * `interface_string` - Index of string descriptor describing this interface
+
+    pub fn interface2(
```
Comment (ianrrees) [inline src/descriptor.rs:202]: I'd prefer a more descriptive name, maybe `interface_alternate()`?

Comment (redpfire) [reply src/descriptor.rs:202]: Right, a more descriptive name than `interface2` would be fitting.

Comment (kiffie) [reply src/descriptor.rs:202]: What about the original one: ```interface_with_alternate_setting```?

Comment (redpfire) [reply src/descriptor.rs:202]: Sounds good but it may be too long.

Comment (redpfire) [reply src/descriptor.rs:202]: Sounds good to me. Ian?

Comment (kiffie) [reply src/descriptor.rs:202]: I change the code. What do you mean with 'lan'?

Comment (redpfire) [reply src/descriptor.rs:202]: What i meant is if Ian has any suggestions. Anyway the change sounds good to me :)

**Comment by redpfire** [comment]
Could you please squash those two commits into one? It will be cleaner after the merge.

**Comment by kiffie** [comment]
Done.

**Comment by ianrrees** [comment]
We should bump the minor version in Cargo.toml as well, since this adds to the API.

**Comment by redpfire** [comment]
Let's bring @mvirkkunen into the discussion.

File: src/descriptor.rs (hunk 1 above)
Comment (mvirkkunen) [reply src/descriptor.rs]: `interface_ex`? Too Win32?

**Comment by mvirkkunen** [comment]
I think this is an OK solution with the current crate version. It should be sufficient for the most important usecases (DFU, USB audio).

File: src/descriptor.rs (hunk 1 above)
Comment (mvirkkunen) [reply src/descriptor.rs]: Perhaps even `interface_alt` :thinking:

**Pull request merged**